### ISSUE TITLE
Please update the images periodicaly

### ISSUE THREAD

**Issue by reidrac** [openSUSE/vagrant#11]
I'm trying to use 42.2 and it is very useful to have these images in Atlas; but currently it is too old and it requires update (and reboot, there are kernel updates) and that breaks provisioning.

Would it be possible to rebuild these images periodically and push them to Atlas if there are changes?

Thanks a lot for providing the images!

**Comment by hwoarang** [comment]
This is not a problem but a feature request :) I currently use vagrant images myself so I plan to update them once a month. However, I can't make long term commitments. Right now, there is a blocking issue (http://bugzilla.suse.com/show_bug.cgi?id=1043758) for me which prevents me from updating the image. Once it's resolved I will do a fresh build.

**Comment by reidrac** [comment]
Thanks!

Perhaps openSUSE infrastructure could be used to automate that.

**Comment by hwoarang** [comment]
@reidrac sure but it needs a volunteer to drive it forward :)

**Comment by denisok** [comment]
@hwoarang I can setup jenkins job on our community [ci](http://storage-ci.suse.de/) to build images periodically. What I would need are credentials to upload them to cloud.

If you could provide that I will build the job. I think 15 and Tumbleweed is just enough?

**Comment by denisok** [comment]
I was able to generate images with this simple script:
```
zypper in -y qemu-kvm libvirt-daemon-qemu libvirt libvirt-devel virtualbox virtualbox-host-source kernel-devel gcc gcc-c++ make

ssh-keygen -t rsa -f /root/.ssh/id_rsa -P ""
cat ~/.ssh/id_rsa.pub >> ~/.ssh/authorized_keys

usermod -a -G vboxusers $USER

systemctl enable libvirtd
systemctl start libvirtd

/sbin/vboxconfig

git clone --depth 1 git@github.com:openSUSE/vagrant.git

cd vagrant

wget https://releases.hashicorp.com/packer/1.2.5/packer_1.2.5_linux_amd64.zip
unzip ./packer*

./packer build -parallel=false -only=virtualbox,libvirt -var 'headless=true' definitions/15.0-x86_64.json
```
and these changes:
https://github.com/openSUSE/vagrant/pull/33

**Comment by aspiers** [comment]
ICBW but I would have thought it would make the most sense to build these automatically in OBS using Kiwi, rather than externally via Packer.  That way the image will be automatically rebuilt and published whenever any of its dependencies gets updated.  Obviously this "push" model of initiating rebuilds will be quicker than polling, and saves having to reinvent the wheel by building custom Jenkins jobs to do the same thing.  It also avoids the requirement for separate CI workers to boot up Packer VMs, since the work would get done by the OBS worker cloud.

This is certainly possible (I've done it) but unfortunately it appears currently undocumented, so I've filed https://github.com/SUSE/kiwi/issues/792 to try to get that fixed.

I suspect there may be other advantages to using Kiwi over Packer, since Kiwi was designed natively for building openSUSE images.  But I don't know the details.  Perhaps @schaefi can comment on that.

**Comment by denisok** [comment]
@aspiers yes, I could build vagrant boxes for vagrant, example is this [internal](https://build.suse.de/package/show/Devel:Storage:6.0/sle15) link.

I don't understand how it would allow us to skip CI, as you need to rung vagrant with just built box. Could you please elaborate?

As for packer vs kiwi, I really don't care - I see the value of kiwi, but untill it is done packer works just fine now. And small jenkins job would do it very fast and simple.

As for kiwi - I would agree that could be better solution but it is not there yet.

**Comment by aspiers** [comment]
[**@denisok**](https://github.com/denisok) commented on [30 Jul 2018, 11:39 BST](https://github.com/openSUSE/vagrant/issues/11#issuecomment-408821594 "2018-07-30T10:39:27Z - Replied by Github Reply Comments"):
> @aspiers yes, I could build vagrant boxes for vagrant, example is this [internal](https://build.suse.de/package/show/Devel:Storage:6.0/sle15) link.
> 
> I don't understand how it would allow us to skip CI, as you need to rung vagrant with just built box. Could you please elaborate?

Ah OK, yes if you want to do the sensible thing and actually validate that the box you just built works then you are right, it doesn't allow you to skip that :-)  I was only talking about skipping the task of building the box, not testing it.  So if you use OBS for that, then you only have to build CI to test it, not any CI for building it first.  The CI could test the latest version from OBS and then if that works, "promote" it into a more stable repository. 

> As for packer vs kiwi, I really don't care - I see the value of kiwi, but untill it is done packer works just fine now. And small jenkins job would do it very fast and simple.
> 
> As for kiwi - I would agree that could be better solution but it is not there yet.

Yeah I don't really know for sure - deferring to @schaefi who might be able to list more compelling reasons to use KIWI over Packer.

**Comment by denisok** [comment]
yep, there are couple of image sources in openSUSE:
https://build.opensuse.org/package/show/Virtualization:Appliances:Images/dice-image-vagrant-libvirt
by @schaefi
https://build.opensuse.org/package/show/home:jloehel:vagrant:images/tumbleweed
by @jloehel

I actually used @jloehel templates before - they are working fine.

The only thing I am missing in @schaefi image project is Static link.

It might be a good idea to create some new vagrant images under:
https://build.opensuse.org/project/show/Virtualization:Appliances:Images
and somehow update https://app.vagrantup.com/opensuse after successful build.

**Comment by aspiers** [comment]
Good news - [building Vagrant images via Kiwi has now been documented](https://github.com/SUSE/kiwi/pull/795).

**Comment by aspiers** [comment]
Documentation now visible: http://suse.github.io/kiwi/building/working_with_images/vagrant_setup.html

**Comment by hwoarang** [comment]
@denisok what credentials do you need to upload them? If you are referring to Atlas, then you need to create an account there and then @vuntz can add you to the necessary group like he did for me. After that you would be able to upload files to the openSUSE organization. Would that work?

**Comment by denisok** [comment]
@hwoarang yes - thanks!

**Comment by AkihiroSuda** [comment]
What's current status?

**Comment by denisok** [comment]
@AkihiroSuda I hope I would have some time later this week to create a job for packer. As for KIWI - that probably longer term, we need someone to do it, so far I don't have time for that.

**Comment by hemna** [comment]
Still no 15.1 box up on vagrant

**Comment by denisok** [comment]
@hemna there were some discussions in SUSE to build it with help of kiwi on OBS, but might be ppl just haven't done it yet.

**Comment by dcermak** [comment]
The openSUSE Tumbleweed box is being periodically built on OBS and linked to Vagrant Cloud with the recipes that are submitted in #40. We are currently working on getting a Leap 15.1 and 15.2 box in the same fashion.
